dandi/tests/test_models.py

```diff
@@ -154,3 +166,192 @@ def test_autogenerated_titles():
+
+
+def test_dantimeta_1():
+    """ checking basic metadata for publishing"""
+    # meta data without doi, datePublished and publishedBy
+    meta_dict = {
+        "identifier": "DANDI:912",
+        "id": "DANDI:912/draft",
+        "version": "v.1",
+        "name": "testing dataset",
+        "description": "testing",
+        "contributor": [
+            {
+                "name": "last name, first name",
+                "roleName": [RoleType("dandi:ContactPerson")],
+            }
+        ],
+        "license": [LicenseType("spdx:CC-BY-4.0")],
+    }
+
+    # should work for DandisetMeta but PublishedDandisetMeta should raise an error
+    DandisetMeta(**meta_dict)
+    with pytest.raises(Exception) as exc:
+        PublishedDandisetMeta(**meta_dict)
+
+    assert [el["msg"] == "field required" for el in exc.value.errors()]
+    assert set([el["loc"][0] for el in exc.value.errors()]) == {
+        "datePublished",
+        "publishedBy",
+        "doi",
+    }
+
+    # after adding doi, datePublished, publishedBy, PublishedDandisetMeta should work
+    meta_dict["doi"] = "00000"
+    meta_dict["datePublished"] = str(datetime.now().year)
+    meta_dict["publishedBy"] = "https://doi.test.datacite.org/dois"
+    PublishedDandisetMeta(**meta_dict)
+
+
+@pytest.mark.parametrize(
+    "additional_meta, datacite_checks",
+    [
+        # no additional meta
+        ({}, {"creators": (1, {"name": "A_last, A_first"})}),
+        # additional contributor with dandi:Author
+        (
+            {
+                "contributor": [
+                    {
+                        "name": "A_last, A_first",
+                        "roleName": [RoleType("dandi:ContactPerson")],
+                    },
+                    {"name": "B_last, B_first", "roleName": [RoleType("dandi:Author")]},
+                ],
+            },
+            {
+                "creators": (1, {"name": "B_last, B_first"}),
+                "contributors": (
+                    1,
+                    {"name": "A_last, A_first", "contributorType": "ContactPerson"},
+                ),
+            },
+        ),
+    ],
+)
+def test_dantimeta_datacite(schema, additional_meta, datacite_checks):
+    """ checking datacite objects for specific metadata dictionaries"""
+
+    prefix = "10.80507"
+    version = "v.0"
+    dandi_id = f"DANDI:{random.randrange(100, 999)}"
+
+    # meta data without doi, datePublished and publishedBy
+    meta_dict = {
+        "identifier": dandi_id,
+        "id": f"{dandi_id}/draft",
+        "version": version,
+        "name": "testing dataset",
+        "description": "testing",
+        "contributor": [
+            {
+                "name": "A_last, A_first",
+                "roleName": [RoleType("dandi:ContactPerson")],
+            }
+        ],
+        "license": [LicenseType("spdx:CC-BY-4.0")],
+        "publishedBy": "https://doi.test.datacite.org/dois",
+        "datePublished": str(datetime.now().year),
```

Comment:
i think i include datetime in datePublished in the model, and should be in isoformat with timezone info.

Reply:
so I should be using `Publishable` class or some version of `DandisetMeta`, e.g. `PublishedDandisetMeta`?

Reply:
attn @djarecka

Reply:
@satra - sorry, forgot to answer, but looks like `str(datetime.now().year)` works fine - do you expect that the error should be raised?